pageserver/src/tenant/timeline.rs

```diff
@@ -3302,14 +2979,15 @@ impl Timeline {
 
         let first_level0_delta = level0_deltas_iter.next().unwrap();
         let mut prev_lsn_end = first_level0_delta.layer_desc().lsn_range.end;
-        let mut deltas_to_compact = vec![Arc::clone(first_level0_delta)];
+        let mut deltas_to_compact = Vec::with_capacity(level0_deltas.len());
+        deltas_to_compact.push(first_level0_delta.guard_against_eviction(true).await?);
         for l in level0_deltas_iter {
             let lsn_range = &l.layer_desc().lsn_range;
 
             if lsn_range.start != prev_lsn_end {
                 break;
             }
-            deltas_to_compact.push(Arc::clone(l));
+            deltas_to_compact.push(l.guard_against_eviction(true).await?);
```

Comment:
This is now the "download l0s" phase (was previously in a loop).

Reply:
> This is now the "download l0s" phase (was previously in a loop).

Good you mentioned it, while reviewing this piece of code the `true` was too subtle.
See my other issue about turning the `true` into an enum, so the callsites become expressive.

Reply:
So this would be much better with `Yes` than `true`.. I don't see it :) I'll add the two different variant's instead of yet another boolean wrapper.

Reply:
d40b9a515 added `Layer::download_and_keep_resident` which is now used for this.